### PR TITLE
Add multi role to user

### LINKED ISSUE #2455: Multiple roles per user
Is there any way to have users with 2 or 3 roles?

### PR DESCRIPTION
**Pull request by gianfigueroa**
⚠️ Add multi role to user 

 I arrange the relationship so that a user has several roles and I modify the midelware to validate by the roles that the user has. 

#### Description:
Arrange the relationship so that a user has several roles and I modify the midelware to validate by the roles that the user has.

#### My PR is a:
- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix #2455
- [ ] 💅 Enhancement
- [x] 🚀 New feature

#### Main update on the:
- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [x] Plugin

#### Manual testing done on the following databases:
- [x] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/config/policies/permissions.js [hunk 1]
```diff
@@ -42,18 +38,33 @@ module.exports = async (ctx, next) => {
...
+      controller: route.controller,
+      action: route.action,
+      enabled: true
+    }, []);
+    if (permission) {
+      bandera = true;
+      break;
+    }
+  }
+  if (bandera == false) {
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/config/policies/permissions.js]: Assuming you mean flag or something of the sort here. You may want to make sure you are using english variable names here.

File: packages/strapi-plugin-users-permissions/config/policies/permissions.js [hunk 2]
```diff
@@ -42,18 +38,33 @@ module.exports = async (ctx, next) => {
...
+    }
+    permission = await strapi.query('permission', 'users-permissions').findOne({
+      role: role._id || role.id,
+      type: route.plugin || 'application',
+      controller: route.controller,
+      action: route.action,
+      enabled: true
+    }, []);
+    if (permission) {
+      bandera = true;
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/config/policies/permissions.js]: English variable names :wink:

**Comment by Aurelsicoko** [comment]
LGTM, @alexandrebodin can you confirm too? 👍

**Comment by alexandrebodin** [comment]
Hum I think this will break stuff as is. To allow such a PR we need tests to be added

**Comment by derrickmehaffy** [comment]
@gianca970226 Can you please run manual tests on all the databases and/or add in automated testing using the e2e scripts for TravisCI?

File: packages/strapi-plugin-users-permissions/config/policies/permissions.js [hunk 3]
```diff
@@ -42,18 +38,33 @@ module.exports = async (ctx, next) => {
...
-    role: role._id || role.id,
-    type: route.plugin || 'application',
-    controller: route.controller,
-    action: route.action,
-    enabled: true
-  }, []);
+  let flag = false;
+  let permission;
+  for (const role of roles) {
+    if (role.type === 'root') {
```
Comment (quadgod) [inline packages/strapi-plugin-users-permissions/config/policies/permissions.js]: maybe better to move role types to constants?

**Comment by lauriejim** [comment]
Hello! Tests are not working.
Can you check this please @giancafigueroa

**Comment by giancafigueroa** [comment]
I hope at night recheck

**Comment by lauriejim** [comment]
I close this PR since we haven't any update.
We will re-open it if it's needed.

**Comment by baermathias** [comment]
@gianfigueroa any news on this feature?

**Pull request closed without merging**